doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -883,14 +781,129 @@ the conversion in hypotheses :n:`{+ @ident}`.
       proof term. See `#12200
       <https://github.com/coq/coq/issues/12200>`_ for more details.
 
-Conversion tactics applied to hypotheses
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. _controlling-the-reduction-strategies:
+
+Controlling the reduction strategies and the conversion algorithm
+----------------------------------------------------------------------
+
+Coq provides reduction strategies that the tactics can invoke and two
+different algorithms to check the convertibility of types. The first
+conversion algorithm lazily compares applicative terms while the other
+is a brute-force but efficient algorithm that first normalizes the
+terms before comparing them. The second algorithm is based on a
+bytecode representation of terms similar to the bytecode
+representation used in the ZINC virtual machine :cite:`Leroy90`. It is
+especially useful for intensive computation of algebraic values, such
+as numbers, and for reflection-based tactics. The commands to fine-
+tune the reduction strategies and the lazy conversion algorithm are
```

Comment:
Except the reduction strategies `vm_compute` and `native_compute` should not be grouped with the other reduction strategies here

Reply:
Because they don't honor Opaque, right?    So don't say "three algorithms" here at all?

Reply:
Would you respond?  Note that I moved this to the beginning of the `Applying conversion rules` section.

Reply:
Sorry, I missed this.  Indeed they don't honor opaque, and also there's not even a way in which they could possibly honor strategy levels other than Opaque and transparent.    I'm not sure what you mean by "three algorithms"

Reply:
I think I picked that up from this wording in the original version.  How about if I refer to "three implementations" instead?

```
.. tacv:: vm_compute
   :name: vm_compute

   This tactic evaluates the goal using the optimized call-by-value evaluation
   bytecode-based virtual machine described in :cite:`CompiledStrongReduction`.
   This algorithm is dramatically more efficient than the algorithm used for the
   :tacn:`cbv` tactic, but it cannot be fine-tuned.
```